Fix various problems in the eunit suite

Vagabond opened the pull request:
The test suite passes consistently for me now, 15 times in a row.

Vagabond committed:
Fix a bunch of broken unit tests

Vagabond committed:
Fix RTQ test

Vagabond committed:
Quiet

Vagabond committed:
Workaround eunit bug with caching funs

See http://erlang.org/pipermail/erlang-bugs/2013-May/003606.html

Vagabond committed:
Make tests more resilant to other tests not cleaning up, fix typos

Vagabond committed:
Fix repl_leader_eqc test to start ranch on the slave nodes

lordnull committed:
Fixed some of rtq eqc test

Fixed a next state splitwith fun; it wasn't updated to use the model's
record format.
Fixed expectation from test's deliver fun; it was not taking the meta into
account.

lordnull committed:
Fixed rtq eqc test double-trim.

lordnull committed:
Fixed extracting seq from model consumer queued item.

lordnull committed:
Fixed delivery of already routed items in model.

lordnull committed:
Removed unused function

A relic from a bygone rebase.

lordnull committed:
Fixed pre-mature dropping of objects.

Pre-mature in that the objects were not added to consumer models' outbound
queue.

lordnull committed:
logic error.

lists:splitwith/2 splits until false, not until true...

lordnull committed:
Fixed postcondition when pulling unroutables.

lordnull committed:
Postcondition now truly respects skipped objects.

lordnull committed:
Fixes for push, pull, and skipping, both tests and rtq.

lordnull committed:
Fixed trim in model on consumer replace.

lordnull committed:
Removed many debug statements and _'ed unused vars.

lordnull committed:
Splits the eqcs to individual tests.

This means if one passes, but the other doesn't, we can easily tell which
one was at fault.

lordnull committed:
Adjusting cached queue size when removing undeliverables

lordnull committed:
Fixed a couple warnings

Unused var and unused function.

Vagabond committed:
Fix a race condition in cluster_mgr get_ipaddrs test

Vagabond committed:
Tweak timeouts

Vagabond committed:
Disable cluster_mgr multinode tests because they don't work everywhere

bookshelfdave commented on src/riak_repl_stats.erl:
```diff
@@ -396,35 +396,43 @@ test_populate_stats() ->
     ReportMinutes = 10,
     for_n_minutes(ReportMinutes, Now, Bytes),
+    %% I think the cache is getting in the way here?
+    timer:sleep(1000),
```
would some type of wait() be more appropriate here?

bookshelfdave commented:
Is this output expected: 
https://gist.github.com/metadave/3f8f5bae407ed061fc3a
and after merging your branch to master (locally):
https://gist.github.com/metadave/233125176758cb67c1fd

bookshelfdave commented:
can anything be done about the huge binaries that meck spits out upon shutdown?

lordnull committed:
Fixed tests reliance on core_ring

Turns out if one thing relies on core_ring_events and
core_ring_manager(test), anything else that needs them will crash unless
It is also started.

bookshelfdave commented:
With @lordnull's latest patch: `All 140 tests passed.`
